browser-extension/mv2/src/modules/EventActions.js

```diff
@@ -0,0 +1,48 @@
+const EventActions = {};
+
+EventActions.sendExtensionEvents = async () => {
+  if (!BG.isAppOnline) {
+    return;
+  }
+
+  let ackReceived = false;
```

Comment:
no need to save this as a variable, 

we can just break the loop based on the response from `sendMessageToApp`